### PR TITLE
Add optional Additional Expense to case contact form

### LINKED ISSUE #2760: Add optional Additional Expense to case contact form
**What type(s) of user does this feature affect?**
 - volunteers

**Description**
As a Volunteer, I would like the New Case Contact form to be organized in a way that is easier for me to understand so that I can input the data that is needed for each case.

**Screenshots of current behavior, if any**  
You can paste images on the clipboard here

**How Should it Work**
- Create New Contact form should look like [mock-up] in link below
  - https://www.figma.com/proto/HBcfH4G9W2pD8psX67orhg/CASA-Redesigns?page-id=0%3A1&node-id=5%3A10&viewport=345%2C48%2C0.5&scaling=min-zoom&starting-point-node-id=5%3A10&show-proto-sidebar=1
  - Other expenses should be associated with feature toggle in System Settings

**Acceptance Criteria**
- Create New Contact form looks like mockup
- Other expenses is displayed when feature toggle is enabled and NOT displayed when it is disbaled

_QA Login Details:_  
[Link to QA site](https://casa-qa.herokuapp.com/)  

Login Emails: 
- volunteer1@example.com  view site as a volunteer
- supervisor1@example.com view site as a supervisor
- casa_admin1@example.com view site as an admin

password for all users: 123456  

### Questions? Join Slack!

We highly recommend that you join us in slack https://rubyforgood.herokuapp.com/ #casa channel to ask questions quickly and hear about office hours (currently Wednesday 6-8pm Pacific), stakeholder news, and upcoming new issues.

### PR DESCRIPTION
**Pull request by DrewAPeterson7671**
### What github issue is this PR for, if any?
Resolves #2760 

### How will this affect user permissions?
- Volunteer permissions:
- Supervisor permissions:
- Admin permissions:

### How is this tested? (please write tests!) 💖💪
rspec

### Feedback please? (optional)
We are very interested in your feedback! Please give us some :) https://forms.gle/1D5ACNgTs2u9gSdh9

### CODE REVIEW

**Comment by DrewAPeterson7671** [comment]
Thank you!  These changes make a lot of sense.  All the research I did never took me in the direction of `ActiveRecord::Base.transaction do`
in the controller.

File: app/models/casa_admin.rb [hunk 1]
```diff
@@ -53,5 +53,5 @@ def deactivate
 #
 # Foreign Keys
 #
-#  fk_rails_...  (casa_org_id => casa_org.id)
+#  fk_rails_...  (casa_org_id => casa_orgs.id)
```
Comment (compwron) [inline app/models/casa_admin.rb:56]: this is an unwanted change, how did this get here?

File: spec/system/case_contacts/new_spec.rb [hunk 2]
```diff
@@ -258,6 +258,12 @@ def index_of(text)
       fill_in "Occurred at", with: "04/04/2020"
       fill_in "Miles driven", with: "30"
       select "Yes", from: "Want driving reimbursement"
+
+      # DP_Edit_Here
```
Comment (compwron) [inline spec/system/case_contacts/new_spec.rb:262]: nit: remove extra comment

**Review by compwron — APPROVED** [review]

**Comment by compwron** [comment]
🎉🎉🎉🎉

File: app/models/casa_admin.rb (hunk 1 above)
Comment (DrewAPeterson7671) [reply app/models/casa_admin.rb:56]: I'm not sure, but I did it last week according to a git tracking extension I got a few days ago.  Its way off of any code I was working with.  Best guess is I clicked on the page at random, maybe while googling something, and hit backspace or delete.

File: spec/system/case_contacts/new_spec.rb (hunk 2 above)
Comment (DrewAPeterson7671) [reply spec/system/case_contacts/new_spec.rb:262]: That's better!

**Comment by DrewAPeterson7671** [comment]
Very nice!  Thank you so much!

**Pull request merged**